test/registration/registration_api_test.rb

```diff
@@ -38,7 +38,7 @@ def test_host_register_template
   end
 
   def test_host_register_template_with_args
-    stub_request(:post, "#{@foreman_url}/register?host=test.example.com").to_return(body: 'template')
+    stub_request(:post, "#{@foreman_url}/register").to_return(body: 'template')
```

Comment:
this test was always wrong, as it turned out, rack test methods defaults to url encoded meaning the data is passed in the body and should not be passed as the query string, this is what this PR implements, therefore changing it the stub this test calls

adding also a test for JSON branch